### ISSUE TITLE
Can't pull EE from private container registry

### ISSUE THREAD

**Issue by mooky31** [ansible/awx#12118]
##### ISSUE TYPE
 - Bug Report
 
##### SUMMARY
Can't pull EE from private container registry : can't validate certificate signed by private authority.
On Registry credentials, Verify SSL in unchecked.

##### ENVIRONMENT
* AWX version: 20.0.1
* AWX install method: operator
* AWX deployment target: minikube
* Operating System: RockyLinux 8
* Web Browser: Firefox

##### STEPS TO REPRODUCE
Try to pull EE from a private registry, with certificates signed by a private authority

##### EXPECTED RESULTS
EE pulled succesfully

##### ACTUAL RESULTS
pod/automation-job-30-h9hpg                            Failed to pull image "registry.local/sdc-sasi/awx:latest": rpc error: code = Unknown desc = Error response from daemon: Get "https://registry.bureautique.itm.lan/v2/": x509: certificate signed by unknown authority

##### ADDITIONAL INFORMATION
I also tried to add the private CA to the host system, didn't change anything.

**Comment by coreydeli** [comment]
This is the same as this issue https://github.com/ansible/awx/issues/12117

**Comment by mooky31** [comment]
I don't think this is the same : the problem here is that the "Verify SSL" tickbox doesn't work

**Comment by AlanCoding** [comment]
The EE `verify_ssl` is implemented in ansible-runner

https://github.com/ansible/ansible-runner/blob/88130033e6b4791952236a8b1eb6f7ac823649e3/test/integration/containerized/test_container_management.py#L114

However, in openshift, it's not the podman CLI but openshift itself that pulls the image. This may be a fairly fundamental issue with how we build the pod spec for container_group jobs. Thanks for the report.

**Comment by shanemcd** [comment]
I'm not sure if we can really do anything about this for container groups. This might be helpful: https://medium.com/@j.wahidin/kubernetes-day-2-operation-working-with-self-signed-image-registry-bd3c7ff1498d

**Comment by valkiriaaquatica** [comment]
> ##### ISSUE TYPE
> * Bug Report
> 
> ##### SUMMARY
> Can't pull EE from private container registry : can't validate certificate signed by private authority. On Registry credentials, Verify SSL in unchecked.
> 
> ##### ENVIRONMENT
> * AWX version: 20.0.1
> * AWX install method: operator
> * AWX deployment target: minikube
> * Operating System: RockyLinux 8
> * Web Browser: Firefox
> 
> ##### STEPS TO REPRODUCE
> Try to pull EE from a private registry, with certificates signed by a private authority
> 
> ##### EXPECTED RESULTS
> EE pulled succesfully
> 
> ##### ACTUAL RESULTS
> pod/automation-job-30-h9hpg Failed to pull image "registry.local/sdc-sasi/awx:latest": rpc error: code = Unknown desc = Error response from daemon: Get "https://registry.bureautique.itm.lan/v2/": x509: certificate signed by unknown authority
> 
> ##### ADDITIONAL INFORMATION
> I also tried to add the private CA to the host system, didn't change anything.

Did you finally solve it? If so, how?

**Comment by mooky31** [comment]
It was a long time ago, so I am not sure.
But if I remember well I configured the CA used to sign the registry's certificate to be trusted by the kubernetes host, and it did the trick. That's not really "Ignore certificate verification" tough...

**Comment by valkiriaaquatica** [comment]
> It was a long time ago, so I am not sure. But if I remember well I configured the CA used to sign the registry's certificate to be trusted by the kubernetes host, and it did the trick. That's not really "Ignore certificate verification" tough...

Yes, I tried that, adding the .crt on the cluster nodes but nothing happened... also tried the tickbox of skip ssl, but also doesn't work  . In my case is nginx ingress on Harbor private repo :(

**Comment by mooky31** [comment]
https://docs.podman.io/en/v2.2.0/markdown/podman-pull.1.html#cert-dir-path

Maybe you should put your certs in /etc/containers/certs.d

**Comment by valkiriaaquatica** [comment]
> https://docs.podman.io/en/v2.2.0/markdown/podman-pull.1.html#cert-dir-path
> 
> Maybe you should put your certs in /etc/containers/certs.d

Hey, thanks! But i'm using rke2 with containerd not podman ,.. I'll keep trying and let here if I find a solution

**Comment by valkiriaaquatica** [comment]
Fnally just ended the problem, was an internal problem with the CA certificate when using the Cloudflare tunnel , no AWX problem at all

> > https://docs.podman.io/en/v2.2.0/markdown/podman-pull.1.html#cert-dir-path
> > Maybe you should put your certs in /etc/containers/certs.d
> 
> Hey, thanks! But i'm using rke2 with containerd not podman ,.. I'll keep trying and let here if I find a solution
